hub/api/dataset.py

```diff
@@ -210,19 +213,26 @@ def __setattr__(self, name: str, value):
         else:
             return super().__setattr__(name, value)
 
-    def __iter__(self):
-        for i in range(len(self)):
-            yield self[i]
```

Comment:
does normal getitem have a lot of overhead due to tensor initialization which is why we're using this syntax? If so add a task to backlog for this

Reply:
It allows prefetching the chunk id and local sample index via iteration instead of `get()` which saves us binary searches down the line when user calls `.numpy()` over the sub tensors.